### DISCUSSION TITLE
The current connection holds no SharePoint context. Please use one of the Connect-PnPOnline commands which uses the -Url argument to connect.

### DISCUSSION THREAD

**Discussion by tldtech** [pnp/powershell#2465] [Q&A]
I have an Azure Runbook with a system managed identity. I'm using the following code as a simple test:
```
Connect-PnPOnline -ManagedIdentity
Get-PnPTenantSite -IncludeOneDriveSites
```
The "Connect-PnPOnline -ManagedIdentity" works fine and connects. However, the next line, "Get-PnPTenantSite -IncludeOneDriveSites" returns the following error:
_"The current connection holds no SharePoint context. Please use one of the Connect-PnPOnline commands which uses the -Url argument to connect."_

If I switch the runbook back to using a "runas" account, it works fine. 
If I try and use "Connect-PnPOnline -ManagedIdentity -Url $url" it complains that there are too many arguments in the command. Is there something different I need to do with using a managed identity?

**Comment by kkazala** [comment]
What works for me:
```bash
$resourceURI ="https://<yourtenant>.sharepoint.com"
$siteUrl="https://<yourtenant>.sharepoint.com/sites/<yoursite>"
# Get token
$tokenAuthURI = $env:IDENTITY_ENDPOINT + "?resource=$resourceURI&api-version=2017-09-01"
$tokenResponse = Invoke-RestMethod -Method Get -Headers @{"Secret" = "$env:IDENTITY_HEADER" } -Uri $tokenAuthURI
$accessToken = $tokenResponse.access_token

# Authenticate to SPO site
Connect-PnPOnline -Url  $siteUrl -AccessToken $accessToken
Get-PnPSite
```

`$env:MSI_SECRET` and `$env:IDENTITY_HEADER` are the same. And so are `$env:MSI_ENDPOINT` and `$env:IDENTITY_ENDPOINT`


[Connect to Azure services in app code](https://learn.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Cpowershell#connect-to-azure-services-in-app-code)

**Comment by chartaam** [comment]
We are having the same problem. (Note that kkazala's suggestion doesn't help -- we are trying to use ManagedIdentity, which is a different way of authentication from using the AccessToken. Not using the ManagedIdentity at all, is not a way to make using the ManagedIdentity work.)

I want to connect using the Managed Identity, then use PNP cmdlets on Sharepoint content.

Has anyone had success using the -ManagedIdentity?

$Conn = Connect-PnPOnline -ManagedIdentity
will connect, but subsequent calls to other PNP cmdlets fail with "The current connection holds no SharePoint context. Please use one of the Connect-PnPOnline commands which uses the -Url argument to connect."

$Conn = Connect-PnPOnline -ManagedIdentity -URL $SiteURL
Fails with "System.Management.Automation.ParameterBindingException: Parameter set cannot be resolved using the specified named parameters."

$Conn = Connect-PnPOnline -ManagedIdentity -ReturnConnection $true
Fails with the same error.

**Reply by kkazala** [reply]
But… it does help 
I am using Managed Identity. I have no app registrations 
The $env: variables are from the context of my Azure Function. 
@chartaam did you at least try?

**Comment by chartaam** [comment]
@kkazala we want to use the -ManagedIdentity parameter, not the -AccessToken parameter. If it turns out that ManagedIdentity doesn't work in the current release, we will give your work-around a try! Thanks.

**Reply by kkazala** [reply]
#2249

**Comment by faisaleem-na** [comment]
no update on this, i am also experiencing the same issue and are trying to use the managed identity?

**Comment by latinus1985** [comment]
Same behaviour. I'm using app registration with cert path and running my code in Azure Pipeline.
